sdk/rust/oak/src/io/receiver.rs

```diff
@@ -66,14 +66,29 @@ impl<T: Decodable> Receiver<T> {
     }
 
     /// Wait for a value to be available from this receiver.
+    ///
+    /// Returns [`ChannelReadStatus`] of the wrapped handle, or `Err(OakStatus::ErrTerminated)` if
+    /// the Oak Runtime is terminating,
     #[allow(clippy::trivially_copy_pass_by_ref)]
-    pub fn wait(&self) -> Result<(), OakStatus> {
+    pub fn wait(&self) -> Result<ChannelReadStatus, OakStatus> {
         // TODO(#500): Consider creating the handle notification space once and for all in `new`.
         let read_handles = vec![self.handle];
         let mut space = crate::new_handle_space(&read_handles);
         crate::prep_handle_space(&mut space);
         let status =
             unsafe { oak_abi::wait_on_channels(space.as_mut_ptr(), read_handles.len() as u32) };
-        crate::result_from_status(status as i32, ())
+
+        match crate::result_from_status(status as i32, ()) {
+            Err(OakStatus::ErrTerminated) => Err(OakStatus::ErrTerminated),
+            _ => match space
```

Comment:
Shouldn't this pass through also `ErrInvalidArgs` and others? Either way, could you explain why some values are matched and some ignored in a comment, for future reference?